### PR TITLE
Allow "add address" and "add customer" and "edit address" buttons to be handled in modal

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Allow "add address" and "add customer" buttons to be handled in modal
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |  https://github.com/PrestaShop/PrestaShop/issues/10583
| How to test?  | Previously, in migrated page "Create an order" ; "add address" and "add customer" were links, so clicking them would make you exit the page. Now they open modals so you dont exit the page.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17090)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/pages/order/create.js
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/pages/order/create.js]: May be a test for checking `orderPageManager` is not null.

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js]:
```suggestion
      refreshAddressesList: () => this.refreshAddressesList(),
```

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js]:
```suggestion
    this.initAddAddressIframe();
```

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js]:
```suggestion
  initAddAddressIframe() {
```

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js]: 2 spaces for indentation in javascript files :)

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js]:
```suggestion
  refreshAddressesList() {
```

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js:551]: No catch?

File: admin-dev/themes/new-theme/js/pages/order/create/customer-manager.js
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/customer-manager.js]:
```suggestion
    this.initAddCustomerIframe();
```

File: admin-dev/themes/new-theme/js/pages/order/create/customer-manager.js
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/customer-manager.js]:
```suggestion
  initAddCustomerIframe() {
```

File: admin-dev/themes/new-theme/js/pages/order/create/customer-manager.js
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/customer-manager.js]:
```suggestion
          },
        );
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php:310]: Why not using `$request->isXmlHttpRequest()`?

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Address/modal_create_success.html.twig
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Address/modal_create_success.html.twig]:
```suggestion
    if (typeof window.parent.getSummary !== "undefined") {
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Address/modal_create_success.html.twig
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Address/modal_create_success.html.twig]:
```suggestion
    if (typeof window.parent.order_create !== "undefined") {
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig]:
```suggestion
    if (typeof window.parent.setupCustomer !== "undefined") {
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig]:
```suggestion
    if (typeof window.parent.order_create !== "undefined") {
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig]: You should escape customerEmail for js
```suggestion
      $('#customer-search-input', window.parent.document).val('{{ customerEmail|e('js') }}');
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php
Comment (matks) [reply src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php:310]: It would be better, but I cannot guarantee all legacy calls using `submitFormAjax` parameter are proper `proper http request`.

In order to comply 100% with legacy behavior and ensure backward compatibility, I stay with previous implementation rules.

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig]: Nice one 👍

**Comment by matks** [comment]
@PierreRambaud your feedbacks have been handled into commit https://github.com/PrestaShop/PrestaShop/pull/17090/commits/be2fd7530a7110fa245c0049597cb3121383fde7

File: src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php
Comment (PierreRambaud) [reply src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php:310]: :sob:

Comment (matks) [reply src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php:310]: I can add a `@todo PS 8.0.0` if you want 😅

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/modal_create_success.html.twig]: the `customerEmail` must be escape everywhere :D same for `customerId`

**Comment by matks** [comment]
I dont know why but I checked out this branch this morning and I get a `fancybox is not a function` TypeError 😅

**Comment by PierreRambaud** [comment]
> I dont know why but I checked out this branch this morning and I get a `fancybox is not a function` TypeError

TROLOLOLO

**Comment by matks** [comment]
@PierreRambaud Fixed and rebased

**Review by PierreRambaud — DISMISSED** [review]

**Comment by matks** [comment]
@Robin-Fischer-PS rebased ✅

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @matks !

Some bugs and a question : 

-When canceling the add of an address, instead of closing the popin, the popin stays displayed and redirect to address page
pr17090 cancel add address
- After creating the first address of a customer, the address is not displayed
pr17090 blank block
- Edit address should be in modale too, so is "Details" of a customer. Will this be done in another PR ?

Thanks !

**Comment by PierreRambaud** [comment]
@Robin-Fischer-PS It's not a bug, it's a feature!

**Comment by matks** [comment]
> Hi @matks !
> 
> Some bugs and a question :
> 
> -When canceling the add of an address, instead of closing the popin, the popin stays displayed and redirect to address page

Fixed

> * After creating the first address of a customer, the address is not displayed

Fixed

> * Edit address should be in modale too, so is "Details" of a customer. Will this be done in another PR ?

Fixed

Please note that there is a difference in behavior for "create address button":
- if, when you click it, there is no address linked to customer, then after successfull address creation I automatically link this new address to the cart being created in the page (which fixes _After creating the first address of a customer, the address is not displayed_)
- if, when you click it, there are already some addresses available for the customer, I only add the new address into the dropdown, I dont automatically assign it

**Review by atomiix — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @matks 

One bug that I should have seen before (sorry) : 
- When I add an address that is not usable for this order (for example, an address in a country not accepted by the shop), there is an unclear error and the address is not added. In Legacy, the address is added, but there is no carrier shown, blocking the creation of the order. I supposed this is caused by the fact that, on Migrated Create an order page, there is only the usable addresses that are listed, in Legacy, all addresses are listed.
pr17090 invalid address

**Comment by matks** [comment]
> * When I add an address that is not usable for this order (for example, an address in a country not accepted by the shop), there is an unclear error and the address is not added.

I cant reproduce the error.

However indeed the address is not added to dropdown if it is not suitable for this order or the shop settings.
Should I display an error message "Address added successfully however it cannot be used for this order as it does meet the requirements of this cart" ?

It seems a bit over-engineered to me 🤔 but if you tell me to do it, I'll do it

**Comment by matks** [comment]
(adding "waiting for QA" label to get an answer)

**Comment by Robin-Fischer-PS** [comment]
Hi @matks ,

If the "weird" error does not appear anymore, there is indeed no need to display another message. However, since I can't install the PR because of conflict, I cannot try to reproduce it.

**Comment by matks** [comment]
PR rebased, ready to QA when I get the 2 approvals

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by NeOMakinG** [comment]
Can you please resolve the conflict @matks ? Thanks :)

**Comment by matks** [comment]
Rebased

**Review by Progi1984 — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @matks !

Since I can't remember what was said on the following topic, I ping you again : 
"Edit address should be in modale too, so is "Details" of a customer. Will this be done in another PR ?"

Otherwise, this PR is OK for me, so as soon as you answer me, I will be able to QA-approve it :yum:

**Comment by Robin-Fischer-PS** [comment]
As seen with @matks , this will be done in another PR.

Issue is here : https://github.com/PrestaShop/PrestaShop/issues/17223 (point 3).

**Comment by Progi1984** [comment]
Hello @matks, thanks for your contribution, would you mind to rebase your pull request? Thanks

**Comment by matks** [comment]
> Hello @matks, thanks for your contribution, would you mind to rebase your pull request? Thanks

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Pull request merged**
